### PR TITLE
Trends default time period update

### PR DESCRIPTION
**Pull request by chasulin**
Trello: https://trello.com/c/7COWZXOf/613-change-the-trend-defaults-on-mobile-to-180-days

Defaults trends view to 180-days on mobile

### CODE REVIEW

File: src/components/Explore/Explore.tsx [hunk 1]
```diff
@@ -252,13 +255,17 @@ const Explore: React.FunctionComponent<{
...
+    // Resets time period state variable when pathname changes (need to force the reset since the route doesn't change):
+    useEffect(() => {
+      setPeriod(defaultTimePeriod);
+    }, [pathname, defaultTimePeriod]);
+
+    // Resets locations+current metric state variables when pathname changes (need to force the reset since the route doesn't change):
     useEffect(() => {
       setSelectedLocations(initialLocations);
       setCurrentMetric(ExploreMetric.HOSPITALIZATIONS);
-      setPeriod(Period.ALL);
```
Comment (chasulin) [inline src/components/Explore/Explore.tsx:261]: need to separate this out into its own effect, so a change in screen size doesn't reset the selected metric+locations

File: src/common/hooks/useDefaultTrendsPeriod.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  );
+
+  useEffect(() => {
+    if (isMobile) {
+      setDefaultTimePeriod(Period.DAYS_180);
+    } else {
+      setDefaultTimePeriod(Period.ALL);
+    }
+  }, [isMobile]);
+  return defaultTimePeriod;
```
Comment (mikelehen) [inline src/common/hooks/useDefaultTrendsPeriod.ts:22]: I'm not sure you need the `defaultTimePeriod` state value and this `useEffect()` call.

I think you could just do:
```
export default function useDefaultTrendsPeriod(): Period {
  const isMobile = useBreakpoint(800);
  return isMobile ? Period.DAYS_180 : Period.ALL;
}
```

And FWIW at that point there's not that much value to having this `useDefaultTrendsPeriod()` hook that's only used in one place.  I'd probably just use the `useBreakpoint()` hook directly from the explore component. 🤷

**Review by mikelehen — APPROVED** [review]
This looks good, though I am a bit surprised we are changing the period when the screen size changes.  I would have probably only used the screen size to set the initial default, and not changed it afterwards.  E.g.:

```
  const [period, setPeriod] = useState<Period>(isMobile ? Period.DAYS_180 : Period.ALL);
```

But I'm not opposed to having it be more responsive like you did. 🤷

**Comment by chasulin** [comment]
> This looks good, though I am a bit surprised we are changing the period when the screen size changes. I would have probably only used the screen size to set the initial default, and not changed it afterwards. E.g.:
> 
> ```
>   const [period, setPeriod] = useState<Period>(isMobile ? Period.DAYS_180 : Period.ALL);
> ```
> 
> But I'm not opposed to having it be more responsive like you did. 🤷

this^^ is what i originally tried but couldn't get to work. isMobile is false on first load, and without the useEffect it doesn't check for when isMobile turns true. i just put it in a hook/file so as to not crowd this file more. fwiw, i couldnt figure out how to get it to work as intended (only use screen size on initial default) 😕

**Comment by chasulin** [comment]
> 

sweet, going to just merge as is 〰

**Pull request merged**